plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,731 @@
+
+        if config.get('user'):
+            users_path = 'data/sonic-snmp:sonic-snmp/SNMP_SERVER_USER/SNMP_SERVER_USER_LIST'
+            payload = self.build_create_user_payload(config)
+            users_request = {'path': users_path, "method": method, 'data': payload}
+            requests.append(users_request)
+            user_path = 'data/sonic-snmp:sonic-snmp/SNMP_SERVER_GROUP_MEMBER/SNMP_SERVER_GROUP_MEMBER_LIST'
+            payload = self.build_create_group_member_payload(config)
+            users_request = {'path': user_path, "method": method, 'data': payload}
+            requests.append(users_request)
+            group_path = 'data/sonic-snmp:sonic-snmp/SNMP_SERVER_GROUP/SNMP_SERVER_GROUP_LIST'
+            payload = self.build_create_group_name_payload(config)
+            if len(payload) > 0:
+                users_request = {'path': group_path, "method": method, 'data': payload}
+                requests.append(users_request)
+
+        if config.get('view'):
+            views_path = 'data/sonic-snmp:sonic-snmp/SNMP_SERVER_VIEW/SNMP_SERVER_VIEW_LIST'
+            payload = self.build_create_view_payload(config)
+            views_request = {'path': views_path, 'method': method, 'data': payload}
+            requests.append(views_request)
+        
+        return requests
+
+    def build_create_agentaddress_payload(self, config):
+        agentaddress = config.get('agentaddress', None)
+        agentaddress_list = list()
+        payload_url = dict()
+        for conf in agentaddress:
+            agentaddress_dict = dict()
+            agentaddress_dict['ip'] = conf.get('ip')
+            agentaddress_dict['port'] = conf.get('port')
+            agentaddress_dict['interface'] = conf.get('interface')
+            agentaddress_list.append(agentaddress_dict)
+
+        payload_url['SNMP_AGENT_ADDRESS_CONFIG_LIST'] = agentaddress_list
+        return payload_url
+
+    def build_create_community_payload(self, config):
+        community = config.get('community', None)
+        community_list = list()
+        payload_url = dict()
+
+        for conf in community:
+            community_dict = dict()
+            community_dict['index'] = conf.get('name')
+            community_dict['securityName'] = conf.get('group')
+            community_list.append(community_dict)
+        payload_url['SNMP_SERVER_COMMUNITY_LIST'] = community_list
+        return payload_url
+
+    def build_create_engine_payload(self, config):
+        engine = config.get('engine', None)
+        engine_list = list() 
+        payload_url = dict()
+
+        for conf in engine:
+            engine_dict = dict()
+            engine_dict['engine-id'] = conf
+            engine_dict['id'] = "GLOBAL"
+            engine_list.append(engine_dict)
+
+        payload_url['SNMP_SERVER_ENGINE_LIST'] = engine_list
+        return payload_url
+
+    def build_create_user_payload(self, config):
+        # TODO: Still needs work
```

Comment:
Can you elaborate on this `TODO`? 
In general, we shouldn't merge `TODO`s -- if we do need to make an exception, file a Github Issue to make sure it's tracked.

Reply:
Removed.